platform/src/CompositePanel.js

```diff
@@ -15,6 +15,30 @@ class CompositePanel extends Panel {
         });
     }
 
+    /**
+     * Check whether there are saveable changes to a child panels contents.
+     * @override
+     * @returns true if there are changes that can be saved.
+     */
+    canSave(){
+        let childrenCanSave = false;
+
+        this.childPanels.forEach( (cp) => childrenCanSave = childrenCanSave || cp.canSave() );
+
+        return childrenCanSave;
+    }
+
+    /**
+     * @override
+     */
+    save(fileHandler){
+        this.childPanels.forEach( (cp) => { 
```

Comment:
This doesn't look right: save needs to return a promise. I was expecting something more along the lines of

```
return Promise.all(this.childPanels.map(p => if (p.canSave()) { return p.save(fileHandler); } else { return null; });
```

Reply:
Oops yes I didn't notice that it returned a promise -  Fixed as suggested in 844db6729f1a71bab2df9d687c83d266242de68a